pkg/handlers/s3_handler.go

```diff
@@ -156,11 +157,49 @@ func (h *s3Handler) OnAdd(ns string, vc kvcv1.VolumeConfig, controllerRef metav1
 	nodeLabelKey := fmt.Sprintf("%s/%s-%s-%s", kvcv1.GroupName, ns, controllerRef.Name, vc.ID)
 	for _, kvcName := range kvcNames {
 		err := waitForPodSuccess(podClient, kvcName, ns, timeout)
+
 		if err != nil {
+			downloadErrMsg := "error during data download and failed to fetch logs for pod"
+
+			podResource := h.k8sClientset.CoreV1().RESTClient().Get().Namespace(ns).Name(kvcName).Resource("pods")
+			if podResource == nil {
+				return kvcv1.Volume{
+					ID:      vc.ID,
+					Message: fmt.Sprintf("%v [name: %v]: %v", downloadErrMsg, kvcName, "getting pod resource failed"),
+				}
+			}
+
+			logReq := podResource.SubResource("log")
+			if logReq == nil {
+				return kvcv1.Volume{
+					ID:      vc.ID,
+					Message: fmt.Sprintf("%v [name: %v]: %v", downloadErrMsg, kvcName, "getting logs sub-resource failed"),
+				}
+			}
+
+			readCloser, streamErr := logReq.Stream()
+			if readCloser != nil {
```

Comment:
Don't think this guard is required when err == nil. Remove this.

Reply:
@balajismaniam but `err` is not `nil` if we are referring to `err` in line `161`.
Would you explain more please?